packages/core/bootstrap/src/lib/middleware/burst-limit/index.ts

```diff
@@ -69,6 +69,9 @@ export const withBurstLimit =
         observedRequestsInMinute > config.burstCapacity1m * MINUTE_LIMIT_WARMER_BUFFER
         // TODO: determine BATCH_REQUEST_BUFFER dynamically based on (number of batch warmers * 3)
       ) {
+        if (config.enabled) {
```

Comment:
This one will always be true, as it covers both custom rate limit capacity and rate limit tier - we should instead check if the `RATE_LIMIT_CAPACITY` env var is set.